sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -380,6 +380,9 @@ object ColumnPruning extends Rule[LogicalPlan] {
         p
       }
 
+    // Eliminate the Projects with empty projectList
+    case p @ Project(projectList, child) if projectList.isEmpty => child
```

Comment:
Thanks @viirya @cloud-fan !

I am not sure which way is better. 
```scala
case p @ Project(_, l: LeafNode) if !l.isInstanceOf[OneRowRelation] => p 
```
My concern is the above line looks more hacky than the current PR fix.